exercises/dolica_akelloegwel/ex02_oo_basics/src/ShapeSorter.cpp

```diff
@@ -0,0 +1,53 @@
+#include "ShapeSorter.h"
+
+void ShapeSorter::PrintBasedOnType(std::vector<std::reference_wrapper<Shape>> shapes, std::string type)
+{
+	// Print shapes that have a given type
+	for (int i = 0; i < shapes.size(); i++)
+		if (shapes[i].get().GetType().compare(type) == 0)
+			std::cout << shapes[i].get() << std::endl;
+}
+void ShapeSorter::PrintBasedOnNSides(std::vector<std::reference_wrapper<Shape>> shapes, int numSides)
+{
+	// Print shapes that have a given number of sides
+	for (int i = 0; i < shapes.size(); i++)
+		if (shapes[i].get().GetNumSides() == numSides)
+			std::cout << shapes[i].get() << std::endl;
+}
+std::vector<std::reference_wrapper<Shape>> ShapeSorter::SortByAreaDesc(std::vector<std::reference_wrapper<Shape>> shapes)
+{
+	// Create a copy of the shapes vector
+	std::vector<std::reference_wrapper<Shape>> copyShapes(shapes);
+
+	// Create a lambda expression for comparing area values
+	auto areaCompare = [](const Shape &s1, const Shape &s2)
+	{
+		return s2.GetArea() < s1.GetArea() || !(s1.GetArea() < s2.GetArea()) && s1.GetArea() < s2.GetArea();
+	};
+
+	// Sort the copied vector
+	std::sort(copyShapes.begin(), copyShapes.end(), areaCompare);
+
+	return copyShapes;
+}
+std::vector<std::reference_wrapper<Shape>> ShapeSorter::SortByPerimeterDesc(std::vector<std::reference_wrapper<Shape>> shapes)
+{
+	// Create a copy of the shapes vector
+	std::vector<std::reference_wrapper<Shape>> copyShapes(shapes);
+
+	// Create a lambda expression for sorting by perimeter values
+	auto perimCompare = [](const Shape &s1, const Shape &s2)

```

Comment:
Would be better as named free function `bool PerimeterComparitor(const Shape &s1, const Shape &s2)` would improve readability in my opinion.